lucene/core/src/java/org/apache/lucene/search/TopFieldCollector.java

```diff
@@ -237,7 +255,9 @@ public void collect(int doc) throws IOException {
                 } else {
                   collectedAllCompetitiveHits = true;
                 }
-              } else if (totalHitsRelation == Relation.GREATER_THAN_OR_EQUAL_TO) {
+              } else if (totalHitsRelation == Relation.EQUAL_TO) {
```

Comment:
Yes the comments above explain why. We want to update the minimum competitive score if the threshold is reached for the first time here. That's the only case where it matters to call updateMinScore on a non-competitive hits. Otherwise it is expected that the minimum score cannot change.